elixir-cocktails/phoenix/creating-forms-with-no-database-table/post.md

```diff
@@ -0,0 +1,171 @@
+            <p>Oops, something went wrong! Please check the errors below:</p>
+            <ul>
+              <%= for {attr, message} <- f.errors do %>
+                <li><%= humanize(attr) %> <%= message %></li>
+              <% end %>
+            </ul>
+          </div>
+        <% end %>
+
+        <div class= "form-group">
+          <%= label f, :first_name, "First name"%>
+          <%= text_input f, :first_name %>
+        </div>
+        <div class= "form-group">
+          <%= label f, :last_name, "Last name" %>
+          <%= text_input f, :last_name %>
+        </div>
+        <div class= "form-group">
+          <%= label f, :email, "Email" %>
+          <%= email_input f, :email %>
+        </div>
+        <div class= "form-group">
+          <%= label f, :content, "Content" %>
+          <%= textarea f, :content %>
+        </div>
+        <%= submit "Send" %>
+
+      <% end %>
+
+    </div>
+
+We can't see our form yet, because `@contact_form` is undefined. Let's fix that. Paste following in place of default index action of your controller (`web/controllers/page_controller.ex`):
+
+    def index(conn, _params) do
+      contact_form = Form.changeset(%Form{})
+      render(conn, "index.html", contact_form: contact_form)
+    end
+
+In order to refer to our FormTestApp.Form as "Form" we need to alias it.
+Just paste following line in controller:
+
+    alias FormTestApp.Form
+
+Now you can start server and see your simple form. You won't however be able
+to submit
+it yet, as we haven't defined proper route and action for that.
+
+### Providing action for submit
+
+As you can see, the form will be sent as a POST request to our
+applications `page_path`. First we need to provide a proper
+route in our `web/router.ex`. Let's add it to browser
+scope:
+
+    post "/", PageController, :create
+
+The only thing left to do is adding action to controller.
+
+    def create(conn, %{"form" => form_params}) do
+      changeset = Form.changeset(%Form{}, form_params)
+
+      if changeset.valid? do
+        conn
+        |> put_flash(:info, "You have passed all the validations.")
+        |> render "index.html", contact_form: Form.changeset(%Form{})
+      else
+        render conn, "index.html", contact_form: changeset
+      end
+    end
+
+Remember that you need to use `plug :scrub_params` in your controller,
+or else all params will be always validated as present (making
+validation on required params useless). Just paste following line in your
+controller.
+
+    plug :scrub_params, "form" when action in [:create]
+
+If you are creating a real form for your app, you probably want to do
+something with users input once the changeset proves valid. Maybe you
+want to send it as an email, or further process the data provided. As this is
+just a simple example, the only thing we added is a flash message
+informing about changeset being valid.
+And that's it! Our simple form with no database table is ready to use.
+
+## Extra tips
+
```

Comment:
This is all very good, @amaranthrose. Can we, however, add a short tip on how you would use the technique with forms that need to pass some ID as well? I mean, it would not be a contact form but possibly something like /settings/2 where 2 would be ID of the user, i.e. you want to create separate form to change some settings related to user and pass the ID along with the form submittion?